lib/puppet/provider/scheduled_task/taskscheduler_api2.rb

```diff
@@ -12,8 +13,16 @@
   defaultfor :operatingsystem => :windows
   confine    :operatingsystem => :windows
 
+  def task_scheduler_helper
+    resource[:compatibility] == 1 ? PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task : PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2Task
```

Comment:
Not sure this is an elegant or sane solution, because this check will get called every time `task_scheduler_helper` is used, but it's a quick comparison.